### ISSUE TITLE
q_{num}_name doesn't reset when cue doesn't exist

### ISSUE THREAD

**Issue by russellsobey** [bitfocus/companion-module-figure53-qlab-advance#27]
I'm trying to use $(INSTANCENAME:q_{num}_name) to label hot keys that are assigned on the fly in Qlab by entering the cue number that corresponds to a key. This works great, but if a cue is deleted or the cue number is cleared the variable doesn't reset, and continues to display the name of a cue that no longer exists. Can't find a way to reset this without restarting Qlab.

"Cue Number color for background" seems to handle this behavior better.

Ideally the name variable would just be an empty string when the cue number is not found, but resetting so Companion defaults to displaying "$NA" would be acceptable.

**Comment by istnv** [comment]
TLDR; 
Add a button to reset the instance:
- Add internal action to disable the instance.
- Add a second action to enable the instance after 100ms.
This will force a re-sync of the cues. Hopefully this is sufficient for now.

When a cue changes, QLab basically says "Q xxx changed". 
Not what changed, just 'something'. 
Normally this is not a problem, because the module then sends a message back to ask QLab for an update for that cue. This ends up with a lot of messages back and forth.
When a cue is deleted the same message, "Q xxx changed".
But when the module asks for more information, QLab sends **nothing** back, basically ignoring the request. Even an error like 'no such cue' would be more useful.
Due to the volume of traffic, I am still working out a way to determine that a request was ignored.

**Comment by russellsobey** [comment]
Ah that makes sense, seems that's a limitation of QLab's API then... Will test your detection efforts once I can get an updated build.

I had tried the disable/enable instance trick, but get a Javascript error when the disable action runs:
`Uncaught Exception:
TypeError: Cannot read property 'removeAllListeners' of undefined
    at osc.TCPSocketPort.<anonymous> (/Applications/Companion.app/Contents/Resources/app.asar/lib/module/figure53-qlab-advance/qlabfb.js:614:18)
    at emitOne (events.js:116:13)
    at osc.TCPSocketPort.emit (events.js:211:7)
    at Socket.<anonymous> (/Applications/Companion.app/Contents/Resources/app.asar/node_modules/osc/src/platforms/osc-node.js:208:18)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at TCP._handle.close [as _onclose] (net.js:554:12)`

**Comment by istnv** [comment]
Which version are you running? That is an issue should have been fixed weeks ago.

**Comment by russellsobey** [comment]
I'm running the 2.0.0-cb62850-2329 build from just a few days ago

**Comment by istnv** [comment]
I was asking about the version because your error points to a function that should be at line 619, not 614. That fix was never updated into the build. Companion keeps each module at a stable point until  the core devs update to include later patches.
I have a large rewrite with a couple of new features (including actual detection of deleted cues) almost ready, but not quite.
I will give the devs a heads up to get the current fixes included.

**Comment by istnv** [comment]
Thanks @haakonnessjoen! 
@russellsobey The latest build should allow the disable/enable workaround to work until I finish the next release.

**Comment by russellsobey** [comment]
Thanks guys, keep up the good work! Will keep an eye out for your new patches as they are merged in...
